Kinvey.Core/Auth/User.cs

```diff
@@ -1149,7 +1142,34 @@ static private LoginToTempURLRequest BuildMICLoginToTempURL(AbstractClient cli,
 			return loginTemp;
 		}
 
-		private LogoutRequest buildLogoutRequest()
+        // Generates a request to login a user to the MIC URL (Resource Owner Credentials Grant (Username and Password))
+        static private MicLoginRequest BuildMICLogin(AbstractClient cli, string username, string password, string clientID)
+        {
+            //    	client_id:  provided by the developer when configuring authlink (defaults to the app’s appKey (the KID))
+            //    	client_secret: App Secret
+            //      grant_type: "password" - this is always set to this value
+            //    	username
+            //    	password
+
+            var appSecret = ((KinveyClientRequestInitializer)cli.RequestInitializer).AppSecret;
+
+            var data = new Dictionary<string, string>
+            {
+                { "client_id", clientID },
+                { "client_secret", appSecret },
```

Comment:
@romanogolikhin I don't think you need the client secret

Reply:
@vinaygahlawat @heyzooi 
I made such way according to the task description
( https://kinvey.atlassian.net/browse/MLIBZ-2524).

You can find this sentence
"The client_id and client_secret values are also required."